### PR TITLE
Fixed looping timer

### PR DESCRIPTION
**Pull request by rphmeier**
Pulling the patch described in https://github.com/ethcore/parity/pull/3592 upstream.

### CODE REVIEW

**Comment by rphmeier** [comment]
Hi, just wondering if the mio team has seen this at all? I really don't like to maintain alternative forks of repos for parity which is why I'm trying to upstream this fix. Any response would be appreciated.

File: src/timer.rs [hunk 1]
```diff
@@ -225,9 +225,13 @@ impl<T> Timer<T> {
         self.poll_to(target_tick)
     }
 
-    fn poll_to(&mut self, target_tick: Tick) -> Option<T> {
+    fn poll_to(&mut self, mut target_tick: Tick) -> Option<T> {
         trace!("tick_to; target_tick={}; current_tick={}", target_tick, self.tick);
 
+		if target_tick < self.tick {
```
Comment (arthurprs) [inline src/timer.rs]: Nit: indentation is off.

**Comment by arthurprs** [comment]
LGTM but I wonder if this is band aiding it instead of fixing the root cause.

**Comment by alexcrichton** [comment]
Ah sorry this slipped under the radar! I merged this as https://github.com/carllerche/mio/commit/ef182bae193a9c7457cd2cf661fcaffb226e3eef and did a whitespace fixup last night, so closing.

**Pull request closed without merging**